lib/compilers/glimmer/index.js

```diff
@@ -9,14 +14,31 @@ export default class GlimmerTemplatePrecompiler extends Filter {
     this.extensions = ['hbs'];
     this.targetExtension = 'ts';
     this.rootName = options.rootName;
+    this.glimmerEnv = options.GlimmerENV || {};
+    this.plugins = options.plugins || { ast: [] };
+
+    if (isFeatureFlagEnabled(this.glimmerEnv.FEATURES, 'glimmer-custom-component-manager')) {
+      this.registerPlugin(ExtractPragmaPlugin);
+    }
+  }
+
+  registerPlugin(plugin) {
+    this.plugins.ast = this.plugins.ast || [];
+    this.plugins.ast.push(plugin);
   }
 
   processString(content, relativePath) {
     let specifier = getTemplateSpecifier(this.rootName, relativePath);
-    return `export default ${this.precompile(content, { meta: { specifier, '<template-meta>': true } })};`;
+    let meta = { specifier, managerId: undefined };
+
+    let precompiledContent = this.precompile(content, { meta })
+
+    return `export default ${precompiledContent};`;
   }
 
   precompile(content, options) {
+    options.plugins = this.plugins;
```

Comment:
We might want to avoid mutating the passed in object:

```js
precompile(content, _options) {
  let plugins = this.plugins;
  let options = Object.assign({}, _options, { plugins });
  return precompile(content, options);
}
```